### PR TITLE
Add lua example

### PR DESCRIPTION
**Pull request by keynmol**

### CODE REVIEW

File: example-lua/src/main/scala/Test.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        if nsize == 0.toULong then
+          (!state)._2 = (!state)._2 + osize.toLong
+          scalanative.libc.stdlib.free(ptr)
+          null
+        else
+          (!state)._1 = (!state)._1 + nsize.toLong
+          scalanative.libc.stdlib.realloc(ptr, nsize)
+      }
+    }
```
Comment (keynmol) [inline example-lua/src/main/scala/Test.scala:23]: This is really cool - apparently you have to provide your own allocator for Lua: https://www.lua.org/manual/5.4/manual.html#lua_Alloc

Which means you can can control its allocations and track them.

File: example-lua/src/main/scala/Test.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            lua_pushstring(L, c"incorrect argument")
+            lua_error(L)
+
+          sum = sum + lua_tointegerx(L, i, null).asInstanceOf[Long]
+
+        lua_pushnumber(L, lua_Number(sum))
+
+        1
+      }
+    }
```
Comment (keynmol) [inline example-lua/src/main/scala/Test.scala:46]: Create a function that Lua will understand (this one just sums all the argument it receives)

File: example-lua/src/main/scala/Test.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    val luaCode = c"""
+  for i = 1, 10, 1 do
+    print("1 + 42 + ", i, " = ", scalaSum(1, 42, i))
+  end
+  """
```
Comment (keynmol) [inline example-lua/src/main/scala/Test.scala:52]: Actual Lua program

File: example-lua/src/main/scala/Test.scala [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    val str = luaL_loadstring(state, luaCode)
+
+    lua_pcallk(
+      state,
+      0,
+      -1,
+      0,
+      0L.asInstanceOf[lua_KContext],
+      lua_KFunction(null)
+    )
```
Comment (keynmol) [inline example-lua/src/main/scala/Test.scala:67]: Run the code (all the magic numbers are copied from lua's sources, they're implemented as macros there :-/

File: example-lua/src/main/scala/Test.scala [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    lua_close(state)
```
Comment (keynmol) [inline example-lua/src/main/scala/Test.scala:69]: Shutdown the state - it will trigger deallocation of everything

**Comment by ckipp01** [comment]
fans-react-to-icarly-revival-embracing-interesting-meme

**Pull request merged**